tests/test_fixity.py

```diff
@@ -634,3 +636,75 @@ def test_scanall_if_sort_argument_is_passed(
             params={"username": STORAGE_SERVICE_USER, "api_key": STORAGE_SERVICE_KEY},
         ),
     ]
+
+
+@mock.patch("requests.get")
+def test_main_handles_exception_if_environment_key_is_missing(
+    _get: mock.Mock, mock_check_fixity: List[mock.Mock]
+) -> None:
+    _get.side_effect = mock_check_fixity
+
+    response = fixity.main(["scan", str(uuid.uuid4())])
+
+    assert str(response) == "Missing environment variable: STORAGE_SERVICE_URL"
+    assert isinstance(response, ArgumentError)
+
+
+@mock.patch("requests.get")
+def test_scanall_handles_exception_if_storage_service_raises_exception(
+    _get: mock.Mock, environment: None
+) -> None:
+    _get.side_effect = [
+        mock.Mock(
+            **{
+                "status_code": 401,
+            },
+            spec=requests.Response,
+        )
+    ]
+
+    response = fixity.main(["scanall"])
+
+    assert (
+        str(response)
+        == f'Storage service at "{STORAGE_SERVICE_URL}" failed authentication while requesting AIPs'
+    )
+    assert isinstance(response, StorageServiceError)
+
+
+@mock.patch("requests.get")
+def test_main_verifies_urls_with_trailing_slash(
+    _get: mock.Mock,
+    mock_check_fixity: List[mock.Mock],
+    monkeypatch: pytest.MonkeyPatch,
+) -> None:
+    _get.side_effect = mock_check_fixity
+    aip_id = uuid.uuid4()
+    stream = io.StringIO()
+    monkeypatch.setenv("STORAGE_SERVICE_URL", "http://foo")
```

Comment:
How are you asserting the trailing slash has been added here?

Reply:
I am adding the assertion for it my next commit.